src/pytest_bdd/gherkin_parser.py

```diff
@@ -0,0 +1,170 @@
+import linecache
+import textwrap
+from pathlib import Path
+from typing import List, Optional
+
+from gherkin.errors import CompositeParserException
+from gherkin.parser import Parser
+from gherkin.token_scanner import TokenScanner
+from pydantic import BaseModel, field_validator, model_validator
+
+from . import exceptions
+from .types import STEP_TYPES
+
+
+class Location(BaseModel):
+    column: int
+    line: int
+
+
+class Comment(BaseModel):
+    location: Location
+    text: str
+
+
+class Cell(BaseModel):
+    location: Location
+    value: str
+
+
+class Row(BaseModel):
+    id: str
+    location: Location
+    cells: List[Cell]
+
+
+class DataTable(BaseModel):
+    name: Optional[str] = None
+    location: Location
+    tableHeader: Optional[Row] = None
+    tableBody: Optional[List[Row]] = None
+
+
+class DocString(BaseModel):
+    content: str
+    delimiter: str
+    location: Location
+
+    @field_validator("content", mode="before")
+    def dedent_content(cls, value: str) -> str:
+        return textwrap.dedent(value)
+
+
+class Step(BaseModel):
+    id: str
+    keyword: str
+    keywordType: str
+    location: Location
+    text: str
+    dataTable: Optional[DataTable] = None
+    docString: Optional[DocString] = None
+
+    @field_validator("keyword", mode="before")
+    def normalize_keyword(cls, value: str) -> str:
+        return value.lower().strip()
+
+    @property
+    def given_when_then(self) -> str:
```

Comment:
What’s this about?

Reply:
This is the "Given/When/Then" value of the step, calculated, as the step may be "And", for example.

I think it's referred to in the parser as "type", which I found confusing, especially as type is used to reference other things other than "given/when/then" type of the step.

Happy to use the same name as the parser currently does of "type"

Reply:
yeah I'd find it clear if it was named `type`.
Maybe we should finally make an Enum for the possible values of step types (given, when, then) and return that instead of a mere string

Reply:
let's try to build the Step object only when we know its concrete type, so that we don't need to set it after its instantiation

Reply:
actually, do we have already this information in the `keyword` and `keywordType`? I guess we will have an `"And"` possible value, but that's fine for this class, since this class is just a wrapper around the untyped dicts returned by the gherkin-parser lib.

So I'd say to remove this property, and instead determine the correct step type (give/when/then) only when we create an instance of the Step (the business model one, `pytest_bdd.parser.Step`)

Reply:
For the moment, I've put in the business logic.  Let me know if you have other thoughts (I just want to focus on the parser here, and avoiding making as many changes outside the parser as possible, but I had already thought along the lines of replacing with an enum - great mind think alike!)